typhon/plugins/core.py

```diff
@@ -70,6 +70,33 @@ def __init__(self, channel, address, protocol=None, parent=None):
         # Add listener
         self.add_listener(channel)
 
+    def cast(self, value):
+        """
+        Cast a value to the correct Python type based on ``signal_type``
+
+        If ``signal_type`` is not set, the result of ``ophyd.Signal.describe``
+        is used to determine what the correct Python type for value is. We need
+        to be aware of the correct Python type so that we can emit the value
+        through the correct signal and convert values returned by the widget to
+        the correct type before handing them to Ophyd Signal
+        """
+        # If this is the first time we are receiving a new value note the type
+        # We make the assumption that signals do not change types during a
+        # connection
+        if not self.signal_type:
+            dtype = self.signal.describe()[self.signal.name]['dtype']
+            # Only way this raises a KeyError is if ophyd is confused
+            self.signal_type = _type_map[dtype][0]
```

Comment:
I wonder if in ophyd there should be a suggested type to cast to. Picking the first one works in all cases here, but feels a bit off.

Reply:
Agreed. I'm all for more being available via introspection of `Signal`. At this point I think this is the best I can do.